### PR TITLE
Introduce SPAN_KIND tag

### PR DESCRIPTION
**Pull request by yurishkuro**
@bensigelman @adriancole @slimsag

### CODE REVIEW

File: opentracing/ext/tags.py [hunk 1]
```diff
@@ -25,17 +25,27 @@
...
 
-# PeerHostIPv6 records IP v6 host address of the peer
+# PEER_HOST_IPV6 records IP v6 host address of the peer
 PEER_HOST_IPV6 = 'peer.ipv6'
 
-# PeerPort records port number of the peer
+# PEER_PORT records port number of the peer
 PEER_PORT = 'peer.port'
+
+# SPAN_KIND hints at relationship between spans, like parent/child
```
Comment (bhs) [inline opentracing/ext/tags.py]: nit: "like client/server"?

File: opentracing/ext/tags.py [hunk 2]
```diff
@@ -25,17 +25,27 @@
...
+SPAN_KIND = 'span.kind'
+
+# Marks a span representing the client-side of an RPC or other remote call
+SPAN_KIND_RPC_CLIENT = 1
```
Comment (bhs) [inline opentracing/ext/tags.py]: no strong feelings, but I probably would have done `= "c"` and `= "s"`... incrementally easier to debug, and comparably expensive over the wire.

Comment (yurishkuro) [inline opentracing/ext/tags.py]: ah, fair enough, I was originally going to use auto-enumeration with `range(3)`, but changed to exact constants to be able to write documentation. I'll change to letters.

FWIW, in my tracer implementation these tags will not actually be sent over the wire, instead the tracer intercepts them and uses to decide between ss/sr and cs/cr annotations. The same happens with all of the PEER tags above, they are used to populate Zipkin's internal `Endpoint` structure. We may want to put this into documentation somewhere.

Comment (bhs) [inline opentracing/ext/tags.py]: oh, good to know. good point that impls can decide how to encode/decode these (a nice property) so really the value is not meaningful. I now care less about this if you would rather use ints :)

**Comment by codefromthecrypt** [comment]
+1

**Pull request merged**